src/nemos/regularizer.py

```diff
@@ -279,10 +139,17 @@ class are defined in the `allowed_solvers` attribute.
         "LBFGSB",
     )
 
+    _default_solver = "GradientDescent"
+
     def __init__(
-        self, solver_name: str = "GradientDescent", solver_kwargs: Optional[dict] = None
+            self,
+            regularizer_strength: float = 1.0
```

Comment:
feels strange to have an unused param regularizer_strength. does it look that bad to have that kwarg only for Lasso, Ridge and GroupLasso?

Reply:
That's what I was thinking as well...

I think we just have it for those and then we can parse in the glm model when the solver is instantiated whether that should be an argument passed or not